VultisigApp/VultisigApp/Services/Solana/SolanaService.swift

```diff
@@ -26,29 +33,35 @@ class SolanaService {
             return response.result
             
         } catch {
-            
-            
-            
             print("Error sending transaction: \(error.localizedDescription)")
         }
         return nil
     }
     
-    func getSolanaBalance(coin: Coin) async throws -> (rawBalance: String, priceRate: Double){
+    func getSolanaBalance(coin: Coin) async throws -> (rawBalance: String, priceRate: Double) {
         var rawBalance = "0"
         let priceRateFiat = await CryptoPriceService.shared.getPrice(priceProviderId: coin.priceProviderId)
         
         do {
-            let data = try await Utils.PostRequestRpc(rpcURL: rpcURL, method: "getBalance", params:  [coin.address])
             
-            if let totalBalance = Utils.extractResultFromJson(fromData: data, path: "result.value") as? Int64 {
-                rawBalance = totalBalance.description
-            }            
+            if coin.isNativeToken {
+                
+                let data = try await Utils.PostRequestRpc(rpcURL: rpcURL, method: "getBalance", params: [coin.address])
+                
+                if let totalBalance = Utils.extractResultFromJson(fromData: data, path: "result.value") as? Int64 {
+                    rawBalance = totalBalance.description
+                }
+                
+            } else {
+                
+                rawBalance = try await fetchTokenBalance(for: coin.address, contractAddress: coin.contractAddress)
+                
+            }
         } catch {
             print("Error fetching balance: \(error.localizedDescription)")
```

Comment:
if you handling error just for printing, need to do it on the top level, not in SolanaService

Reply:
I will allow it to throw the error always for all inside the Solana Service.